app/src/main/java/com/github/se/travelpouch/ui/notifications/NotificationItem.kt

```diff
@@ -0,0 +1,226 @@
+      }
+}
+
+@Composable
+fun AcceptButton(
+    notification: Notification,
+    listTravelViewModel: ListTravelViewModel,
+    profileViewModel: ProfileModelView,
+    notificationViewModel: NotificationViewModel,
+    context: android.content.Context
+) {
+  Button(
+      onClick = {
+        handleInvitationResponse(
+            notification,
+            listTravelViewModel,
+            profileViewModel,
+            notificationViewModel,
+            context,
+            isAccepted = true)
+      },
+      modifier = Modifier.padding(end = 8.dp).testTag("notification_item_accept_button"),
+      colors =
+          ButtonDefaults.buttonColors(
+              containerColor = Color.Transparent, contentColor = Color(0xFF12c15d))) {
+        Text(text = "ACCEPT")
+      }
+}
+
+@Composable
+fun DeclineButton(
+    notification: Notification,
+    listTravelViewModel: ListTravelViewModel,
+    profileViewModel: ProfileModelView,
+    notificationViewModel: NotificationViewModel,
+    context: android.content.Context
+) {
+  Button(
+      onClick = {
+        handleInvitationResponse(
+            notification,
+            listTravelViewModel,
+            profileViewModel,
+            notificationViewModel,
+            context,
+            isAccepted = false)
+      },
+      colors =
+          ButtonDefaults.buttonColors(containerColor = Color.Transparent, contentColor = Color.Red),
+      modifier = Modifier.testTag("notification_item_decline_button")) {
+        Text(text = "DECLINE")
+      }
+}
+
+fun handleInvitationResponse(
+    notification: Notification,
+    listTravelViewModel: ListTravelViewModel,
+    profileViewModel: ProfileModelView,
+    notificationViewModel: NotificationViewModel,
+    context: android.content.Context,
+    isAccepted: Boolean
+) {
+  listTravelViewModel.getTravelById(
+      notification.travelUid,
+      { travel ->
+        val responseType = if (isAccepted) NotificationType.ACCEPTED else NotificationType.DECLINED
+        val responseMessage = if (isAccepted) "ACCEPTED" else "DECLINED"
+
+        val invitationResponse =
+            Notification(
+                notification.notificationUid,
+                profileViewModel.profile.value.fsUid,
+                notification.senderUid,
+                notification.travelUid,
+                NotificationContent.InvitationResponseNotification(
+                    profileViewModel.profile.value.name, travel!!.title, isAccepted),
+                responseType)
+
+        notificationViewModel.sendNotification(invitationResponse)
+        if (isAccepted) {
+          listTravelViewModel.addUserToTravel(
+              profileViewModel.profile.value.email,
+              travel,
+              { updatedContainer ->
+                listTravelViewModel.selectTravel(updatedContainer)
+                Toast.makeText(context, "User added successfully!", Toast.LENGTH_SHORT).show()
+              },
```

Comment:
Why are you selecting a travel here ?

Reply:
No, this is not useful here. You should delete this line, since you already call select travel when clicking on a card